tests/topotests/bgp_communities_topo1/test_bgp_communities_topo2.py

```diff
@@ -0,0 +1,713 @@
+        tc_name, result)
+
+    step("Verify that these prefixes, originated on R1, are now"
+         "received on both routers R2 and R3")
+    for addr_type in ADDR_TYPES:
+        input_dict_4 = {
+            "r1": {
+                "static_routes": [
+                    {
+                        "network": NETWORK[addr_type],
+                        "next_hop": topo["routers"]["r2"]["links"]\
+                            ["r1"][addr_type].split("/")[0]
+                    }
+                ]
+            }
+        }
+        result = verify_bgp_rib(tgen, addr_type, "r2", input_dict_4,
+                            next_hop = topo["routers"]["r1"]["links"]\
+                            ["r2"][addr_type].split("/")[0])
+        assert result is True, "Testcase  : Failed \n Error: {}".format(
+            tc_name, result)
+        result = verify_bgp_rib(tgen, addr_type, "r3", input_dict_4,
+                            next_hop = topo["routers"]["r1"]["links"]\
+                            ["r3"][addr_type].split("/")[0])
+        assert result is True, "Testcase  : Failed \n Error: {}".format(
+            tc_name, result)
+
+    step("Create a route-map on R1 to set community as local-AS")
+    seq_id = 10
+    input_rmap = {
+        "r1": {
+            "route_maps": {
+                "rmap_wkc": [{
+                    "action": "permit",
+                    "seq_id": seq_id,
+                            "set": {
+                            "community": {"num": "local-AS"}
+                            }
+                        }]
+                }
+            }
+        }
+    result = create_route_maps(tgen, input_rmap)
+    assert result is True, 'Testcase {} : Failed \n Error: {}'.format(
+        tc_name, result)
+
+    step("Apply route-map while redistributing static routes into BGP")
+    input_dict_2 = {
+            "r1": {
+                "bgp": {
+                "address_family": {
+                    "ipv4": {
+                        "unicast": {
+                            "redistribute": [{
+                                "redist_type": "static",
+                                "attribute": {
+                                    "route-map" : "rmap_wkc"
+                                }
+                            }]
+                        }
+                    },
+                    "ipv6": {
+                        "unicast": {
+                            "redistribute": [{
+                                "redist_type": "static",
+                                "attribute": {
+                                    "route-map" : "rmap_wkc"
+                                }
+                            }]
+                        }
+                    }
+                }
+            }
+        }
+    }
+    result = create_router_bgp(tgen, topo, input_dict_2)
+
+    step("Verify that BGP prefixes on R1 have community: local-AS")
```

Comment:
I see that most of the code is duplicated for each community. Can't be helpers be used here to minimize the code and review it easier?

Reply:
Hi @ton31337 , Thanks for your comment, I have made the changes. Please have a look.